evm/src/executor/inspector/cheatcodes/expect.rs

```diff
@@ -55,16 +58,29 @@ pub fn handle_expect_revert(
         return Err("Call reverted as expected, but without data".to_string().encode().into())
     }
 
-    let (err, actual_revert): (_, Bytes) = match retdata {
-        _ if retdata.len() >= 4 && retdata[0..4] == [8, 195, 121, 160] => {
+    let string_data = match retdata {
+        _ if retdata.len() >= REVERT_PREFIX.len() &&
+            retdata[..REVERT_PREFIX.len()] == REVERT_PREFIX =>
+        {
+            Some(&retdata[4..])
+        }
+        _ if retdata.len() >= ERROR_PREFIX.len() &&
+            &retdata[..ERROR_PREFIX.len()] == ERROR_PREFIX.as_slice() =>
+        {
+            Some(&retdata[ERROR_PREFIX.len()..])
+        }
+        _ => None,
+    };
+
+    let (err, actual_revert): (_, Bytes) = match string_data {
+        Some(data) => {
             // It's a revert string, so we do some conversion to perform the check
-            let decoded_data: Bytes =
-                ethers::abi::decode(&[ethers::abi::ParamType::Bytes], &retdata[4..])
-                    .expect("String error code, but data is not a string")[0]
-                    .clone()
-                    .into_bytes()
-                    .expect("Cannot fail as this is bytes")
-                    .into();
+            let decoded_data: Bytes = ethers::abi::decode(&[ethers::abi::ParamType::Bytes], data)
+                .expect("String error code, but data is not a string")[0]
```

Comment:
I know this is legacy but, I believe you can use `<Bytes as AbiDecode>::decode()` here directly, which get rid of the clone and the additional into_bytes conversion as well .

or `Vec<u8>::decode`, depending on what Bytes type this actually is lol

Reply:
ok it took a while for me to realize that there's actually several `Bytes` types